cmd/frontend/graphqlbackend/repository_comparison_test.go

```diff
@@ -0,0 +1,617 @@
+package graphqlbackend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"html/template"
+	"io"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sourcegraph/go-diff/diff"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/externallink"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/graphqlutil"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/types"
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/gitserver"
+	"github.com/sourcegraph/sourcegraph/internal/highlight"
+	"github.com/sourcegraph/sourcegraph/internal/vcs/git"
+)
+
+func TestRepositoryComparison(t *testing.T) {
+	ctx := context.Background()
+
+	testBaseRevision := "24f7ca7c1190835519e261d7eefa09df55ceea4f"
+	testHeadRevision := "b69072d5f687b31b9f6ae3ceafdc24c259c4b9ec"
+
+	repo := &types.Repo{
+		ID:   api.RepoID(1),
+		Name: api.RepoName("github.com/sourcegraph/sourcegraph"),
+		// If we set this, NewRepositoryComparison needs an external service in
+		// the database.
+		// ExternalRepo: api.ExternalRepoSpec{
+		// 	ID:          "external-id",
+		// 	ServiceType: "github",
+		// 	ServiceID:   "https://github.com/",
+		// },
+	}
+
+	git.Mocks.GetCommit = func(id api.CommitID) (*git.Commit, error) {
+		if string(id) != testBaseRevision && string(id) != testHeadRevision {
+			t.Fatalf("GetCommit received wrong ID: %s", id)
+		}
+
+		return &git.Commit{ID: id}, nil
+	}
+	defer func() { git.Mocks.GetCommit = nil }()
```

Comment:
```suggestion
	t.Cleanup(func() { git.Mocks.GetCommit = nil })
```

Same for other occurrences.